src/locale/keymap-util.c

```diff
@@ -780,3 +781,137 @@ int x11_convert_to_vconsole(Context *c) {
 
         return modified;
 }
+
+bool locale_gen_check_available(void) {
+        struct stat st;
+        if (stat("/usr/sbin/locale-gen", &st) < 0)
+                return false;
+        if (stat("/etc/locale.gen", &st) < 0)
+                return false;
+        return true;
+}
+
+int locale_gen_enable_locale(char *locale[_VARIABLE_LC_MAX]) {
+        _cleanup_fclose_ FILE *fr = NULL;
+        _cleanup_fclose_ FILE *fw = NULL;
+        _cleanup_free_ char *temp_path = NULL;
```

Comment:
use `_cleanup_(unlink_and_freep)` for this please, so that any return path from this function automatically removes the file. And then, after rename() succeeded, invalidate this variable with `temp_path = mfree(temp_path)`, so that we don't make any attempts to remove the file anymore